data/adapters/gersbach_E2G_CRISPR_adapter.py

```diff
@@ -0,0 +1,119 @@
+import csv
+import gzip
+import json
+from typing import Optional
+
+from adapters.helpers import build_regulatory_region_id, parse_guide_file
+from adapters.file_fileset_adapter import FileFileSet
+from adapters.writer import Writer
+
+# Example rows from Gersbach's CRISPR screen data
+# rowID baseMean    log2FoldChange  lfcSE   stat    pvalue  padj
+# AHCTF1_1    403.0544398 -0.430896793    0.483371961 -0.891439364    0.372693508 0.842070692
+# AHCTF1_2  422.638436 -0.109560771 0.517889371 -0.211552461    0.8324562   0.969028334
+# AHCTF1_3  537.6979771 0.231612028 0.431039354 0.537333832 0.59103704  0.932431857
+
+
+class GersbachE2GCRISPR:
+
+    ALLOWED_LABELS = [
+        'genomic_element',
+        'genomic_element_gene'
+    ]
+    SOURCE = 'IGVF'
+
+    def __init__(self, filepath, reference_filepath, label, source_url, writer: Optional[Writer] = None, **kwargs):
+        if label not in GersbachE2GCRISPR.ALLOWED_LABELS:
+            raise ValueError('Ivalid label. Allowed values: ' +
+                             ','.join(GersbachE2GCRISPR.ALLOWED_LABELS))
+        self.data_file = filepath
+        self.guide_file = reference_filepath
+        self.label = label
+        self.source_url = source_url
+        self.file_accession = source_url.split('/')[-2]
+        self.dataset = label
+        self.type = 'edge'
+        if (self.label == 'genomic_element'):
+            self.type = 'node'
+        self.writer = writer
+
+    def process_file(self):
+        self.writer.open()
+        guide_rna_sequences = parse_guide_file(self.guide_file)
+        genomic_elements = {}
+        guide_id_to_element_id = {}
+        for guide_id, guide_rna in guide_rna_sequences.items():
+            name = guide_rna.get('intended_target_name')
+            gene = name.split('.')[0] if name.startswith('ENSG') else None
```

Comment:
can add a gene validation step here (import adapters.gene_validator, like in AFGR_eqtl_adapter.py )

Reply:
added.